make_prg/utils/io_utils.py

```diff
@@ -0,0 +1,221 @@
+import gzip
+import fileinput
+import shutil
+import uuid
+from Bio import AlignIO
+import os
+from loguru import logger
+from make_prg.from_msa import MSA
+from typing import Dict, Optional, List, Tuple
+from collections import defaultdict
+from pathlib import Path
+from zipfile import ZipFile
+
+
+def load_alignment_file(msa_file: [str, Path], alignment_format: str) -> MSA:
+    msa_file = str(msa_file)
+    if msa_file.endswith(".gz"):
+        handle = gzip.open(msa_file, "rt")
+        alignment = AlignIO.read(handle, alignment_format)
+        handle.close()
+    else:
+        alignment = AlignIO.read(msa_file, alignment_format)
+    for record in alignment:
+        record.seq = record.seq.upper()
+    return alignment
+
+
+def concatenate_text_files(input_filepaths, output_filepath):
+    output_filepath_parent_dir = Path(output_filepath).parent
+    os.makedirs(output_filepath_parent_dir, exist_ok=True)
+
+    with open(output_filepath, "w") as fout, fileinput.input(input_filepaths) as fin:
+        for line in fin:
+            fout.write(line)
+
+
+# Note: not unit tested
+# From https://gist.github.com/jacobtomlinson/9031697
+def remove_empty_folders(path: str, remove_root: bool = True):
+    if not os.path.isdir(path):
+        return
+
+    # remove empty subfolders
+    files = os.listdir(path)
+    if len(files):
+        for f in files:
+            fullpath = os.path.join(path, f)
+            if os.path.isdir(fullpath):
+                remove_empty_folders(fullpath)
```

Comment:
```suggestion
    for f in files:
        fullpath = os.path.join(path, f)
        if os.path.isdir(fullpath):
            remove_empty_folders(fullpath)
```

Reply:
Thanks again, no idea why I kept this conditional